### PR TITLE
feat: introduce hardhat-deploy-config 

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Creates a new package hardhat-deploy-config. We're using the same
configuration system for all of our contracts package, so might as well
turn it into a hardhat plugin to avoid duplicating code.

### CODE REVIEW

File: packages/hardhat-deploy-config/.env.example [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# URL for an L1 RPC provider, used to query L2 output proposals
+FAULT_DETECTOR__L1_RPC_PROVIDER=
+
```
Comment (tynes) [inline packages/hardhat-deploy-config/.env.example:3]: This is a dead file

File: packages/hardhat-deploy-config/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  "name": "@eth-optimism/hardhat-deploy-config",
+  "version": "0.1.0",
+  "description": "[Optimism] Hardhat deploy configuration plugin",
+  "main": "dist/src/index.js",
+  "types": "dist/src/index.d.ts",
+  "files": [
+    "dist/*"
+  ],
+  "scripts": {
+    "start": "ts-node ./src/service.ts",
```
Comment (tynes) [inline packages/hardhat-deploy-config/package.json:11]: `start` is dead code?

**Comment by smartcontracts** [comment]
@tynes to get this over the finish line

**Comment by tynes** [comment]
The last major problem that I see here is that we lose the ability to have network specific config. There are a few possibilities on how to fix this, going to think about it a bit

**Review by maurelian — APPROVED** [review]
All looks quite reasonable to me.

**Pull request merged**